Class1/TopKElement.java

```diff
@@ -0,0 +1,48 @@
+package Class1;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.PriorityQueue;
+import java.util.Queue;
+
+// https://leetcode.com/problems/top-k-frequent-elements/
+// Start: 11:03
+// end: 11:11
+public class TopKElement {
```

Comment:
> can you write the N log K solution for me. After that, can you brainstorm a solution better than N log K? One does exist. :slightly_smiling_face: If you get stuck ask for hints here rather than looking online.

Reply:
1. yes N log K solution  is submitted here:  https://github.com/zhanwz/CodingExercise/commit/db1d04821d4b07e87a64c6e638fcfc31f4637b36

Reply:
2. Yes, I have another idea which is similar to quick sort. However the worse case time complexity is O(N^2). Is this an acceptable solution or should I keep thinking?
        // Another idea comes to my mind is sth similar to quick sort:
        // 1. create num to freq hashmap 
        // 2. select a pivot and count for the freq
        // 3. move elements with higher freq on the right
        // 4. if that's no of elements higher than pivot is k then we get result and can exit the recursion.
        // 5. different from quick sort is we only need to keep partition on half of the result
        // Avg time complexity: O (n) + o(n/2) + o(n/4)... keep look at half of the result = O(2n)
        // Worst case time complexity: O(N^2)
        // Space: O(N)